src/integration/data/upsertAndUpdate.test.ts

```diff
@@ -0,0 +1,95 @@
+import { Pinecone, Index } from '../../index';
+import {
+  randomString,
+  generateRecords,
+  INDEX_NAME,
+  sleep,
+  waitUntilRecordsReady,
+} from '../test-helpers';
+
+describe('upsert and update', () => {
+  let pinecone: Pinecone,
+    index: Index,
+    ns: Index,
+    namespace: string,
+    recordIds: string[];
+
+  beforeEach(async () => {
+    pinecone = new Pinecone();
+
+    await pinecone.createIndex({
+      name: INDEX_NAME,
+      dimension: 5,
+      metric: 'cosine',
+      spec: {
+        serverless: {
+          region: 'us-west-2',
+          cloud: 'aws',
+        },
+      },
+      waitUntilReady: true,
+      suppressConflicts: true,
+    });
+
+    namespace = randomString(16);
+    index = pinecone.index(INDEX_NAME);
+    ns = index.namespace(namespace);
+  });
+
+  afterEach(async () => {
+    await ns.deleteMany(recordIds);
+  });
+
+  test('verify upsert and update', async () => {
+    const recordToUpsert = generateRecords(5, 1, false, true);
+    recordIds = recordToUpsert.map((r) => r.id);
+    expect(recordToUpsert).toHaveLength(1);
+    expect(recordToUpsert[0].id).toEqual('0');
+
+    await ns.upsert(recordToUpsert);
+    await waitUntilRecordsReady(ns, namespace, recordIds);
+    await sleep(1500);
+
+    // Fetch and inspect records to validate upsert
+    const fetchResult = await ns.fetch(recordIds);
+    const records = fetchResult.records;
+    if (records) {
+      expect(records['0']).toBeDefined();
+      expect(records['0'].values).toEqual(recordToUpsert[0].values);
+      expect(records['0'].metadata).toEqual(recordToUpsert[0].metadata);
+    } else {
+      fail(
+        'Did not find expected records. Fetch result was ' +
+          JSON.stringify(fetchResult)
+      );
+    }
+
+    const oldMetadata = records['0'].metadata;
+
+    // Update record values
+    const newValues = [0.5, 0.4, 0.3, 0.2, 0.1];
+    const newMetadata = { flavor: 'chocolate' };
+    await ns.update({
+      id: '0',
+      values: newValues,
+      metadata: newMetadata,
+    });
+    await sleep(1500);
```

Comment:
Have you seen freshness perform differently on update vs upsert? In the `waitUntilRecordsReady` function below you are polling on a 3 second interval for new data to appear, so I'm curious is 1500ms is long enough here.

Reply:
Yeah, from testing it seems like update is a much quicker operation than waiting for the initial upsert. However, calling `fetch` immediately after updating returns an unmodified record.

I ran several runs locally and this did fail one time, so I've bumped to 3000ms to stay consistent with other areas.